CDCgov/prime-reportstream
# Fix MSH-2, MSH-3 for ELIMS message 52

This PR makes changes that allow MSH-2 and MSH-3 to be preserved when processing ELIMS messages. MSH-2 is typically either `^~\&` or `^~\$#`, but ReportStream has always reduced that to `^~\&`. These changes allow for a custom extension on the FHIR `MessageHeader` to store the value of the field in cases where it is something other than `^~\&`. Then on conversion from FHIR to HL7v2, that extension can also be used to preserve the value as well. As for MSH-3, that was only being changed due to the default sender transform; overriding the relevant schema elements allowed for that to be resolved very easily.

Test Steps:
1. Update the dev-elims receiver in `organizations.yml` to be active and reload the settings.
2. Send a report to ReportStream from the `ignore-elr-elims` sender using message 52 in [this folder](https://cdc.sharepoint.com/teams/ReportStream/Shared%20Documents/Forms/AllItems.aspx?RootFolder=%2Fteams%2FReportStream%2FShared%20Documents%2FELIMS%2DELR) as the message body.
3. Compare the resulting HL7 to the input HL7 and observe that any differences are appropriately documented in the [notes spreadsheet](https://cdc.sharepoint.com/:x:/r/teams/ReportStream/Shared%20Documents/ELIMS-ELR/52_Message_RS_processing_notes.xlsx)

## Changes
- Adjusted HL7v2->FHIR mapping for MessageHeader to store MSH-2 as a custom extension if not `^~\&`
- Added elements in `elims-sender-transform` to override default transforms for MSH-3
- Adjusted currently unused schemas to use the new format for `valueSet`s (unrelated, but discovered while testing and is an easy fix)
- Added FHIR->HL7v2 mapping to read custom extension for MSH-2
- Renamed `ELRReceiver` to `UniversalPipelineReceiver` (again unrelated, but relatively small changeset to do this while in the area)
- Moved or deleted code that blindly replaces `^~\&#` with `^~\&`. In some cases, this is now only done for legacy pipelines.
- Added code to temporarily replace `^~\&#` with `^~\&` before calling library functions which don't handle `^~\&#` in v2.5.1, and then change back to the original value afterward.

## Checklist

### Testing

## Linked Issues
- Addresses #10204

## To Be Done (before submitting PR)
- [x] Add integration tests
- [x] Fix integration tests to expect MSH-2 to be preserved
- [x] Check code coverage

## To Be Done (generally)
- Get updates from ELIMS team about remaining items from [the spreadsheet](https://cdc.sharepoint.com/:x:/r/teams/ReportStream/_layouts/15/Doc.aspx?sourcedoc=%7BFAED6C19-BA41-4F03-A148-6C8490228A1C%7D&file=HL7%20IN%20vs%20HL7%20OUT%20_%2052_24095_04204575_5120_DIFFERENCES.xlsx&action=default&mobileredirect=true)
- #10733
